### PR TITLE
HLL support default value

### PR DESCRIPTION
**Pull request by HangyuanLiu**
When load date don't have HLL value , HLL value  is a empty HyperLogLog

eg:
CREATE TABLE `test_uv_9` (
  `pin_id` bigint(20) NOT NULL COMMENT "",
  `id` bigint(20) NULL COMMENT "",
  `uv1` hll HLL_UNION NOT NULL COMMENT "",
  `uv2` hll HLL_UNION NOT NULL COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`pin_id`, `id`)
DISTRIBUTED BY HASH(`pin_id`) BUCKETS 16
PROPERTIES (
"storage_type" = "COLUMN"
)
1、curl --location-trusted -u root:123456 -H column_separator:, -H label:test_uv_14  -H "columns:pin_id,idx,u1,u2,id=12 ,uv1=hll_hash(u1)" -T uv_test http://11.40.166.162:8030/api/test/test_uv_9/_stream_load
Result ：hll_union_agg(uv2) should be 0，And hll_union_agg(uv1) has values > 0

2、curl --location-trusted -u root:123456 -H column_separator:, -H label:test_uv_15  -H "columns:pin_id,idx,u1,u2,id=12" -T uv_test http://11.40.166.162:8030/api/test/test_uv_10/_stream_load
Result ：hll_union_agg(uv1) and hll_union_agg(uv2) should be 0

### CODE REVIEW

**Comment by imay** [comment]
@HangyuanLiu 
Can you add some document or usage example for this change?

File: be/src/olap/aggregate_func.h [hunk 1]
```diff
@@ -408,7 +408,11 @@ struct AggregateFuncTraits<OLAP_FIELD_AGGREGATION_HLL_UNION, OLAP_FIELD_TYPE_HLL
         dst_slice->size = sizeof(HyperLogLog);
         // use 'placement new' to allocate HyperLogLog on arena, so that we can control the memory usage.
         char* mem = arena->Allocate(dst_slice->size);
-        dst_slice->data = (char*) new (mem) HyperLogLog(src_slice->data);
+        if (src_slice->empty()) {
+            dst_slice->data = (char*) new (mem) HyperLogLog();
```
Comment (kangkaisen) [inline be/src/olap/aggregate_func.h]: When the slice size is empty?

Comment (HangyuanLiu) [reply be/src/olap/aggregate_func.h]: When HLL columns are NULL in the LOAD data

**Comment by HangyuanLiu** [comment]
Update example @imay

**Comment by kangkaisen** [comment]
> curl --location-trusted -u root:123456 -H column_separator:, -H label:test_uv_15 -H "columns:pin_id,idx,u1,u2,id=12" -T uv_test http://11.40.166.162:8030/api/test/test_uv_10/_stream_load
Result ：hll_union_agg(uv1) and hll_union_agg(uv2) should be 0

I think we don't support this usage.

When load data with HLL column, we must use hll_hash function, hll_hash function will handle null value.

**Comment by HangyuanLiu** [comment]
This PR will support this capability.
There is a scenario where a Doris table may have multiple HLL columns that may come from different streams. The A stream may not contain the required HLL columns in the B stream and may not have NULL fields. The number of fields in this stream may be less than the number of fields in the Doris table
@kangkaisen

**Comment by kangkaisen** [comment]
@HangyuanLiu Hi, I See.
Could we add a `empty_hll` function support this capability? add then, we could still ensure the HLL input in `aggregate_func.h` will not null.

**Comment by HangyuanLiu** [comment]
@kangkaisen 
If we add a empty_hll function , so we should write load command like .

LOAD LABEL test.uv 
DATA INFILE ("hdfs://ns1017/**streamA**/*") INTO TABLE `test_uv`
(pin, id, u1)
set (
  uv1=hll_hash(u1),
  uv2=empty_hash(u2),
);

LOAD LABEL test.uv 
DATA INFILE ("hdfs://ns1017/**streamB**/*") INTO TABLE `test_uv`
(pin, id, u1)
set (
  uv1= empty_hash(u1),
  uv2= hll_hash(u2),
);

uv2 may be another business stream
It is unreasonable to let the import data for A business  care about the other B business HLL type.

**Comment by morningman** [comment]
@HangyuanLiu I think using `empty_hll()` is more reasonable. If business A does not care not column `uv2`, It should leave `uv2` unchanged by explicitly assigning a function means `unchanged`, which is what `empty_hll()` will do.

If user assign nothing to HLL column, HLL column's content will be undefined. Because HLL is not like other types of column, which can be filled by default value or null.

`empty_hll()` to HLL is like `0` to SUM.

And also, should will and a function `empty_bitmap()` for BITMAP column? @kangkaisen

**Comment by kangkaisen** [comment]
> And also, should will and a function empty_bitmap() for BITMAP column

@morningman  I agree with you.

**Comment by imay** [comment]
> > And also, should will and a function empty_bitmap() for BITMAP column
> 
> @morningman I agree with you.

I agree too

@HangyuanLiu 

Will you add a hll_empty() function? And it will be better if you can create an issue to record this improvement.

**Comment by HangyuanLiu** [comment]
OK,I can add a function

File: be/src/exprs/hll_function.cpp [hunk 2]
```diff
@@ -25,6 +25,17 @@ namespace doris {
 using doris_udf::BigIntVal;
 using doris_udf::StringVal;
 
+const std::string HllFunctions::hll_empty_buffer = init_empty_hll();
+std::string HllFunctions::init_empty_hll() {
+    const int HLL_EMPTY_SIZE = 1;
+    std::string buf;
+    std::unique_ptr<HyperLogLog> hll;
+    hll.reset(new HyperLogLog());
```
Comment (imay) [inline be/src/exprs/hll_function.cpp]: A stack variable is enough

File: be/src/olap/hll.h [hunk 3]
```diff
@@ -103,6 +103,14 @@ class HyperLogLog {
 
     int64_t estimate_cardinality();
 
+    std::string empty() {
```
Comment (imay) [inline be/src/olap/hll.h]: why not make this function static?

**Review by imay — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
#1895

**Pull request merged**